apps/api/src/app/workflows-v2/clients/novu-base-client.ts

```diff
@@ -0,0 +1,170 @@
+// Base HttpError class with response field
+export class HttpError extends Error {
+  constructor(
+    public responseText: string,
+    public status: number,
+    public response: Response // Add response field
+  ) {
+    super(`${status}: ${responseText}`);
+    this.name = this.constructor.name;
+  }
+
+  toString(): string {
+    return `${this.name} (status: ${this.status}): ${this.responseText}`;
+  }
+}
+
+// Specific error classes extending HttpError
+export class NovuBadRequestError extends HttpError {}
+export class NovuUnauthorizedError extends HttpError {}
+export class NovuForbiddenError extends HttpError {}
+export class NovuNotFoundError extends HttpError {}
+export class NovuInternalServerError extends HttpError {}
+export class NovuNotImplementedError extends HttpError {}
+export class NovuBadGatewayError extends HttpError {}
+export class NovuServiceUnavailableError extends HttpError {}
+export class Novu extends HttpError {}
+export class NovuRedirectError extends HttpError {
+  redirectUrl: string;
+
+  constructor(responseText: string, status: number, redirectUrl: string, response: Response) {
+    super(responseText, status, response); // Pass response to the base class
+    this.redirectUrl = redirectUrl;
+  }
+}
+
+// Map of status codes to specific error classes
+const errorMap: Record<number, new (responseText: string, status: number, response: Response) => HttpError> = {
+  400: NovuBadRequestError,
+  401: NovuUnauthorizedError,
+  403: NovuForbiddenError,
+  404: NovuNotFoundError,
+  500: NovuInternalServerError,
+  501: NovuNotImplementedError,
+  502: NovuBadGatewayError,
+  503: NovuServiceUnavailableError,
+  504: Novu,
```

Comment:
Per the comment above, it looks like this class should be renamed to describe a `504` exception.

Reply:
fixed it, thank you for the catch